### PR TITLE
Lower terminators.

### PR DESCRIPTION
**Pull request by vext01**
Companion PR for https://github.com/softdevteam/yk/pull/8

WIll need Cargo.lock regeneration after the above is merged.

### CODE REVIEW

**Comment by vext01** [comment]
Now the yk change is in, i've synced the lock file.

OK to squash?

**Comment by ltratt** [comment]
Please squash.

File: src/librustc_yk_sections/emit_tir.rs [hunk 1]
```diff
@@ -110,8 +110,72 @@ impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
...
+                    cleanup_bb: cleanup_bb.map(|bb| u32::from(bb)),
+                    ret_bb: ret_bb,
+                }
+            },
+            TerminatorKind::Assert{target: target_bb, cleanup: cleanup_bb, ..} =>
+                ykpack::Terminator::Assert{
+                    target_bb: u32::from(target_bb),
+                    cleanup_bb: cleanup_bb.map(|bb| u32::from(bb)),
+                },
+            TerminatorKind::Yield{resume: resume_bb, drop: drop_bb, ..} =>
```
Comment (bjorn3) [inline src/librustc_yk_sections/emit_tir.rs]: This and some other terminators do not reach codegen: https://github.com/rust-lang/rust/blob/d96c01e77c9a812ed9f101d0e0714e6bd833a0bf/src/librustc_codegen_ssa/mir/block.rs#L879-L883

Comment (vext01) [reply src/librustc_yk_sections/emit_tir.rs]: Great stuff! Thanks!

**Comment by vext01** [comment]
@ltratt, we might as well go back and address @bjorn3 's comment. So there will be one more ykpack change before we merge this.

**Comment by ltratt** [comment]
Makes sense to me!

**Comment by vext01** [comment]
If you are OK, I'll squash this?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
Splat.

**Comment by vext01** [comment]
Fix pushed, ok to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**